src/options.rs

```diff
@@ -127,11 +127,49 @@ impl Options {
             auth: AuthStyle::Credentials {
                 jwt_cb: {
                     let path = path.as_ref().to_owned();
-                    Arc::new(move || creds_utils::user_jwt_from_file(&path))
+                    Arc::new(move || {
+                        let (jwt, _kp) = auth_utils::load_creds(&path)?;
+                        Ok(jwt)
+                    })
                 },
                 sig_cb: {
                     let path = path.as_ref().to_owned();
-                    Arc::new(move |nonce| creds_utils::sign_nonce_with_file(nonce, &path))
+                    Arc::new(move |nonce| {
+                        let (_jwt, kp) = auth_utils::load_creds(&path)?;
+                        auth_utils::sign_nonce(nonce, kp)
+                    })
+                },
+            },
+            ..Default::default()
+        }
+    }
+
+    /// Authenticate with NATS using a `.nk` file.
+    ///
+    /// # Example
+    /// ```no_run
+    /// # fn main() -> std::io::Result<()> {
+    /// let nc = nats::Options::with_nkey("./path/to/my.nk")
```

Comment:
I think nkeys are usually bundled in or env variables, want to make sure we support that model. Having nkey in a separate file may be the exception.

Reply:
How about the following two models?

* `Options::with_nkey_file("./path/to/my.nk")` - loads the seed key from a file.
* `Options::with_nkey("publickey", || load_seed())` - is given the public key and a function that loads the seed key when needed.

The Go client has `func Nkey(pubKey string, sigCB SignatureHandler) Option` but I don't know why a function that signs the nonce (`SignatureHandler`) is preferable over a function that simply loads the seed key (because we know how to sign the nonce once we have the seed key).

Which signature of these two do you like more?

1. `Options::with_nkey("publickey", || load_seed())`
2. `Options::with_nkey("publickey", |nonce| sign(nonce))`

Reply:
Its a function because with HSM there will be no private key, you have to call out to have it sign for you etc.

Reply:
Okay, I changed `Options::with_nkey()` to take a closure that signs the nonce.